IllegalStateException: ManagedProvider is not available

The following exception is thrown for each of my lifx:colormzlight Things whenever I start the runtime (openHAB build # 1094):

```
21:13:55.145 [ERROR] [core.common.registry.AbstractProvider] - Could not inform the listener 'org.eclipse.smarthome.core.thing.link.ThingLinkManager$3@20f6fb2e' about the 'UPDATED' event!: ManagedProvider is not available
java.lang.IllegalStateException: ManagedProvider is not available
	at org.eclipse.smarthome.core.common.registry.AbstractRegistry.remove(AbstractRegistry.java:226) ~[?:?]
	at org.eclipse.smarthome.core.thing.link.ThingLinkManager$3.updated(ThingLinkManager.java:255) ~[?:?]
	at org.eclipse.smarthome.core.thing.link.ThingLinkManager$3.updated(ThingLinkManager.java:1) ~[?:?]
	at org.eclipse.smarthome.core.common.registry.AbstractProvider.notifyListeners(AbstractProvider.java:57) ~[?:?]
	at org.eclipse.smarthome.core.common.registry.AbstractProvider.notifyListenersAboutUpdatedElement(AbstractProvider.java:81) ~[?:?]
	at org.eclipse.smarthome.core.common.registry.AbstractManagedProvider.update(AbstractManagedProvider.java:127) ~[?:?]
	at org.eclipse.smarthome.core.thing.internal.ThingManager$1$1.run(ThingManager.java:226) ~[?:?]
	at org.eclipse.smarthome.core.thing.internal.ThingManager$1$1.run(ThingManager.java:1) ~[?:?]
	at java.security.AccessController.doPrivileged(Native Method) ~[?:?]
	at org.eclipse.smarthome.core.thing.internal.ThingManager$1.thingUpdated(ThingManager.java:214) ~[?:?]
	at org.eclipse.smarthome.core.thing.binding.BaseThingHandler.updateThing(BaseThingHandler.java:467) ~[?:?]
	at org.eclipse.smarthome.binding.lifx.handler.LifxLightHandler.addRemoveZoneChannels(LifxLightHandler.java:353) ~[?:?]
	at org.eclipse.smarthome.binding.lifx.handler.LifxLightHandler.access$9(LifxLightHandler.java:336) ~[?:?]
	at org.eclipse.smarthome.binding.lifx.handler.LifxLightHandler$CurrentLightState.updateZoneChannels(LifxLightHandler.java:197) ~[?:?]
	at org.eclipse.smarthome.binding.lifx.handler.LifxLightHandler$CurrentLightState.updateColorChannels(LifxLightHandler.java:163) ~[?:?]
	at org.eclipse.smarthome.binding.lifx.handler.LifxLightHandler$CurrentLightState.setColors(LifxLightHandler.java:135) ~[?:?]
	at org.eclipse.smarthome.binding.lifx.internal.LifxLightState.setColor(LifxLightState.java:131) ~[?:?]
	at org.eclipse.smarthome.binding.lifx.internal.LifxLightCurrentStateUpdater.handleLightStatus(LifxLightCurrentStateUpdater.java:165) ~[?:?]
	at org.eclipse.smarthome.binding.lifx.internal.LifxLightCurrentStateUpdater.handleResponsePacket(LifxLightCurrentStateUpdater.java:139) ~[?:?]
	at org.eclipse.smarthome.binding.lifx.internal.LifxLightCommunicationHandler.lambda$2(LifxLightCommunicationHandler.java:369) ~[?:?]
	at java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:890) ~[?:?]
	at org.eclipse.smarthome.binding.lifx.internal.LifxLightCommunicationHandler.lambda$1(LifxLightCommunicationHandler.java:369) ~[?:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:?]
	at java.lang.Thread.run(Thread.java:748) [?:?]
```

As a result the respective thing handlers are not properly initialized so they do not handle commands.

Thanks for already having creating a fix for this issue @kaikreuzer . While testing I've come across a similar exception (openHAB build # 1099). It occurs when starting the runtime while the Inbox has unapproved Things and the  "Auto Approve" setting of the Inbox is enabled:

```
21:55:25.076 [ERROR] [rg.eclipse.smarthome.config.discovery] - [org.eclipse.smarthome.config.discovery.internal.AutomaticInboxProcessor(22)] The activate method has thrown an exception
java.lang.IllegalStateException: ManagedProvider is not available
	at org.eclipse.smarthome.core.common.registry.AbstractRegistry.add(AbstractRegistry.java:213) [109:org.eclipse.smarthome.core:0.9.0.201711242229]
	at org.eclipse.smarthome.config.discovery.internal.PersistentInbox.addThingSafely(PersistentInbox.java:562) [104:org.eclipse.smarthome.config.discovery:0.9.0.201711242229]
	at org.eclipse.smarthome.config.discovery.internal.PersistentInbox.approve(PersistentInbox.java:171) [104:org.eclipse.smarthome.config.discovery:0.9.0.201711242229]
	at org.eclipse.smarthome.config.discovery.internal.AutomaticInboxProcessor.approveAllInboxEntries(AutomaticInboxProcessor.java:206) [104:org.eclipse.smarthome.config.discovery:0.9.0.201711242229]
	at org.eclipse.smarthome.config.discovery.internal.AutomaticInboxProcessor.activate(AutomaticInboxProcessor.java:218) [104:org.eclipse.smarthome.config.discovery:0.9.0.201711242229]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:?]
	at org.apache.felix.scr.impl.inject.BaseMethod.invokeMethod(BaseMethod.java:229) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.inject.BaseMethod.access$500(BaseMethod.java:39) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.inject.BaseMethod$Resolved.invoke(BaseMethod.java:650) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.inject.BaseMethod.invoke(BaseMethod.java:506) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.inject.ActivateMethod.invoke(ActivateMethod.java:307) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.inject.ActivateMethod.invoke(ActivateMethod.java:299) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.manager.SingleComponentManager.createImplementationObject(SingleComponentManager.java:298) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.manager.SingleComponentManager.createComponent(SingleComponentManager.java:109) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.manager.SingleComponentManager.getService(SingleComponentManager.java:906) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.manager.SingleComponentManager.getServiceInternal(SingleComponentManager.java:879) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.manager.AbstractComponentManager.activateInternal(AbstractComponentManager.java:749) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.manager.DependencyManager$SingleStaticCustomizer.addedService(DependencyManager.java:1012) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.manager.DependencyManager$SingleStaticCustomizer.addedService(DependencyManager.java:968) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.manager.ServiceTracker$Tracked.customizerAdded(ServiceTracker.java:1215) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.manager.ServiceTracker$Tracked.customizerAdded(ServiceTracker.java:1136) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.manager.ServiceTracker$AbstractTracked.trackAdding(ServiceTracker.java:945) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.manager.ServiceTracker$AbstractTracked.track(ServiceTracker.java:881) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.manager.ServiceTracker$Tracked.serviceChanged(ServiceTracker.java:1167) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.BundleComponentActivator$ListenerInfo.serviceChanged(BundleComponentActivator.java:127) [42:org.apache.felix.scr:2.0.12]
	at org.eclipse.osgi.internal.serviceregistry.FilteredServiceListener.serviceChanged(FilteredServiceListener.java:109) [?:?]
	at org.eclipse.osgi.internal.framework.BundleContextImpl.dispatchEvent(BundleContextImpl.java:915) [?:?]
	at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:230) [?:?]
	at org.eclipse.osgi.framework.eventmgr.ListenerQueue.dispatchEventSynchronous(ListenerQueue.java:148) [?:?]
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.publishServiceEventPrivileged(ServiceRegistry.java:862) [?:?]
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.publishServiceEvent(ServiceRegistry.java:801) [?:?]
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistrationImpl.register(ServiceRegistrationImpl.java:127) [?:?]
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.registerService(ServiceRegistry.java:225) [?:?]
	at org.eclipse.osgi.internal.framework.BundleContextImpl.registerService(BundleContextImpl.java:464) [?:?]
	at org.apache.felix.scr.impl.manager.AbstractComponentManager$3.register(AbstractComponentManager.java:887) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.manager.AbstractComponentManager$3.register(AbstractComponentManager.java:874) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.manager.RegistrationManager.changeRegistration(RegistrationManager.java:132) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.manager.AbstractComponentManager.registerService(AbstractComponentManager.java:941) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.manager.AbstractComponentManager.activateInternal(AbstractComponentManager.java:741) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.manager.DependencyManager$SingleStaticCustomizer.addedService(DependencyManager.java:1012) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.manager.DependencyManager$SingleStaticCustomizer.addedService(DependencyManager.java:968) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.manager.ServiceTracker$Tracked.customizerAdded(ServiceTracker.java:1215) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.manager.ServiceTracker$Tracked.customizerAdded(ServiceTracker.java:1136) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.manager.ServiceTracker$AbstractTracked.trackAdding(ServiceTracker.java:945) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.manager.ServiceTracker$AbstractTracked.track(ServiceTracker.java:881) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.manager.ServiceTracker$Tracked.serviceChanged(ServiceTracker.java:1167) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.BundleComponentActivator$ListenerInfo.serviceChanged(BundleComponentActivator.java:127) [42:org.apache.felix.scr:2.0.12]
	at org.eclipse.osgi.internal.serviceregistry.FilteredServiceListener.serviceChanged(FilteredServiceListener.java:109) [?:?]
	at org.eclipse.osgi.internal.framework.BundleContextImpl.dispatchEvent(BundleContextImpl.java:915) [?:?]
	at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:230) [?:?]
	at org.eclipse.osgi.framework.eventmgr.ListenerQueue.dispatchEventSynchronous(ListenerQueue.java:148) [?:?]
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.publishServiceEventPrivileged(ServiceRegistry.java:862) [?:?]
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.publishServiceEvent(ServiceRegistry.java:801) [?:?]
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistrationImpl.register(ServiceRegistrationImpl.java:127) [?:?]
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.registerService(ServiceRegistry.java:225) [?:?]
	at org.eclipse.osgi.internal.framework.BundleContextImpl.registerService(BundleContextImpl.java:464) [?:?]
	at org.apache.felix.scr.impl.manager.AbstractComponentManager$3.register(AbstractComponentManager.java:887) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.manager.AbstractComponentManager$3.register(AbstractComponentManager.java:874) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.manager.RegistrationManager.changeRegistration(RegistrationManager.java:132) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.manager.AbstractComponentManager.registerService(AbstractComponentManager.java:941) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.manager.AbstractComponentManager.activateInternal(AbstractComponentManager.java:741) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.manager.DependencyManager$SingleDynamicCustomizer.addedService(DependencyManager.java:799) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.manager.DependencyManager$SingleDynamicCustomizer.addedService(DependencyManager.java:743) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.manager.ServiceTracker$Tracked.customizerAdded(ServiceTracker.java:1215) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.manager.ServiceTracker$Tracked.customizerAdded(ServiceTracker.java:1136) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.manager.ServiceTracker$AbstractTracked.trackAdding(ServiceTracker.java:945) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.manager.ServiceTracker$AbstractTracked.track(ServiceTracker.java:881) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.manager.ServiceTracker$Tracked.serviceChanged(ServiceTracker.java:1167) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.BundleComponentActivator$ListenerInfo.serviceChanged(BundleComponentActivator.java:127) [42:org.apache.felix.scr:2.0.12]
	at org.eclipse.osgi.internal.serviceregistry.FilteredServiceListener.serviceChanged(FilteredServiceListener.java:109) [?:?]
	at org.eclipse.osgi.internal.framework.BundleContextImpl.dispatchEvent(BundleContextImpl.java:915) [?:?]
	at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:230) [?:?]
	at org.eclipse.osgi.framework.eventmgr.ListenerQueue.dispatchEventSynchronous(ListenerQueue.java:148) [?:?]
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.publishServiceEventPrivileged(ServiceRegistry.java:862) [?:?]
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.publishServiceEvent(ServiceRegistry.java:801) [?:?]
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistrationImpl.register(ServiceRegistrationImpl.java:127) [?:?]
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.registerService(ServiceRegistry.java:225) [?:?]
	at org.eclipse.osgi.internal.framework.BundleContextImpl.registerService(BundleContextImpl.java:464) [?:?]
	at org.apache.felix.scr.impl.manager.AbstractComponentManager$3.register(AbstractComponentManager.java:887) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.manager.AbstractComponentManager$3.register(AbstractComponentManager.java:874) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.manager.RegistrationManager.changeRegistration(RegistrationManager.java:132) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.manager.AbstractComponentManager.registerService(AbstractComponentManager.java:941) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.manager.AbstractComponentManager.activateInternal(AbstractComponentManager.java:741) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.manager.AbstractComponentManager.enableInternal(AbstractComponentManager.java:675) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.manager.AbstractComponentManager.enable(AbstractComponentManager.java:430) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.manager.ConfigurableComponentHolder.enableComponents(ConfigurableComponentHolder.java:657) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.BundleComponentActivator.initialEnable(BundleComponentActivator.java:341) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.Activator.loadComponents(Activator.java:390) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.Activator.access$200(Activator.java:54) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.scr.impl.Activator$ScrExtension.start(Activator.java:265) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.utils.extender.AbstractExtender.createExtension(AbstractExtender.java:254) [42:org.apache.felix.scr:2.0.12]
	at org.apache.felix.utils.extender.AbstractExtender.modifiedBundle(AbstractExtender.java:227) [42:org.apache.felix.scr:2.0.12]
	at org.osgi.util.tracker.BundleTracker$Tracked.customizerModified(BundleTracker.java:482) [?:?]
	at org.osgi.util.tracker.BundleTracker$Tracked.customizerModified(BundleTracker.java:415) [?:?]
	at org.osgi.util.tracker.AbstractTracked.track(AbstractTracked.java:232) [?:?]
	at org.osgi.util.tracker.BundleTracker$Tracked.bundleChanged(BundleTracker.java:444) [?:?]
	at org.eclipse.osgi.internal.framework.BundleContextImpl.dispatchEvent(BundleContextImpl.java:903) [?:?]
	at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:230) [?:?]
	at org.eclipse.osgi.framework.eventmgr.ListenerQueue.dispatchEventSynchronous(ListenerQueue.java:148) [?:?]
	at org.eclipse.osgi.internal.framework.EquinoxEventPublisher.publishBundleEventPrivileged(EquinoxEventPublisher.java:213) [?:?]
	at org.eclipse.osgi.internal.framework.EquinoxEventPublisher.publishBundleEvent(EquinoxEventPublisher.java:120) [?:?]
	at org.eclipse.osgi.internal.framework.EquinoxEventPublisher.publishBundleEvent(EquinoxEventPublisher.java:112) [?:?]
	at org.eclipse.osgi.internal.framework.EquinoxContainerAdaptor.publishModuleEvent(EquinoxContainerAdaptor.java:156) [?:?]
	at org.eclipse.osgi.container.Module.publishEvent(Module.java:476) [?:?]
	at org.eclipse.osgi.container.Module.start(Module.java:467) [?:?]
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.incStartLevel(ModuleContainer.java:1620) [?:?]
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.incStartLevel(ModuleContainer.java:1600) [?:?]
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.doContainerStartLevel(ModuleContainer.java:1571) [?:?]
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.dispatchEvent(ModuleContainer.java:1514) [?:?]
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.dispatchEvent(ModuleContainer.java:1) [?:?]
	at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:230) [?:?]
	at org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.run(EventManager.java:340) [?:?]
```

--- a/bundles/core/org.eclipse.smarthome.core.thing.test/src/test/groovy/org/eclipse/smarthome/core/thing/internal/ThingManagerOSGiTest.groovy
+++ b/bundles/core/org.eclipse.smarthome.core.thing.test/src/test/groovy/org/eclipse/smarthome/core/thing/internal/ThingManagerOSGiTest.groovy
@@ -123,6 +123,9 @@ class ThingManagerOSGiTest extends OSGiTest {
         waitForAssert {
             assertThat getBundleContext().getServiceReferences(ReadyMarker, "(" + ThingManager.XML_THING_TYPE + "=" + getBundleContext().getBundle().getSymbolicName() + ")"), is(notNullValue())
         }
+        waitForAssert {
+            assertThat getBundleContext().getServiceReferences(ChannelItemProvider, null), is(notNullValue())
+        }
     }
 
     @After
@@ -684,6 +687,7 @@ class ThingManagerOSGiTest extends OSGiTest {
             }
         ] as ThingHandlerFactory
         registerService(thingHandlerFactory)
+        waitForAssert { assertThat itemRegistry.get(itemName), is(notNullValue()) }
 
         callback.statusUpdated(THING, ThingStatusInfoBuilder.create(ThingStatus.ONLINE).build())
 
@@ -1568,6 +1572,7 @@ class ThingManagerOSGiTest extends OSGiTest {
         registerService(thingHandlerFactory)
 
         itemChannelLinkRegistry.add(new ItemChannelLink("testItem", new ChannelUID(THING.getUID(), "channel")))
+        waitForAssert { assertThat itemRegistry.get("testItem"), is(notNullValue()) }
 
         eventPublisher.post(ItemEventFactory.createCommandEvent("testItem", new StringType("TEST")))
 

--- a/bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/link/ItemChannelLinkRegistry.java
+++ b/bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/link/ItemChannelLinkRegistry.java
@@ -130,7 +130,7 @@ public class ItemChannelLinkRegistry extends AbstractLinkRegistry<ItemChannelLin
         this.itemRegistry = null;
     }
 
-    @Reference(cardinality = ReferenceCardinality.OPTIONAL, policy = ReferencePolicy.DYNAMIC, name = "ManagedItemChannelLinkProvider")
+    @Reference
     protected void setManagedProvider(ManagedItemChannelLinkProvider provider) {
         super.setManagedProvider(provider);
     }

--- a/bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/link/ManagedItemChannelLinkProvider.java
+++ b/bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/link/ManagedItemChannelLinkProvider.java
@@ -51,7 +51,7 @@ public class ManagedItemChannelLinkProvider extends DefaultAbstractManagedProvid
         }
     }
 
-    @Reference(policy = ReferencePolicy.DYNAMIC)
+    @Reference(policy = ReferencePolicy.STATIC)
     @Override
     protected void setStorageService(StorageService storageService) {
         super.setStorageService(storageService);

--- a/bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/link/ManagedItemChannelLinkProvider.java
+++ b/bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/link/ManagedItemChannelLinkProvider.java
@@ -19,7 +19,6 @@ import org.eclipse.smarthome.core.storage.StorageService;
 import org.eclipse.smarthome.core.thing.ThingUID;
 import org.osgi.service.component.annotations.Component;
 import org.osgi.service.component.annotations.Reference;
-import org.osgi.service.component.annotations.ReferencePolicy;
 
 /**
  *
@@ -51,7 +50,7 @@ public class ManagedItemChannelLinkProvider extends DefaultAbstractManagedProvid
         }
     }
 
-    @Reference(policy = ReferencePolicy.STATIC)
+    @Reference
     @Override
     protected void setStorageService(StorageService storageService) {
         super.setStorageService(storageService);

--- a/bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/link/ThingLinkManager.java
+++ b/bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/link/ThingLinkManager.java
@@ -63,9 +63,9 @@ public class ThingLinkManager extends AbstractTypedEventSubscriber<ThingStatusIn
 
     private static final String THREADPOOL_NAME = "thingLinkManager";
 
-    private Logger logger = LoggerFactory.getLogger(ThingLinkManager.class);
+    private final Logger logger = LoggerFactory.getLogger(ThingLinkManager.class);
 
-    private final ScheduledExecutorService scheduler = ThreadPoolManager.getScheduledPool("thingLinkManager");
+    private final ScheduledExecutorService scheduler = ThreadPoolManager.getScheduledPool(THREADPOOL_NAME);
 
     private ThingRegistry thingRegistry;
     private ManagedThingProvider managedThingProvider;

--- a/bundles/model/org.eclipse.smarthome.model.thing.tests/src/org/eclipse/smarthome/model/thing/tests/GenericItemChannelLinkProviderJavaTest.java
+++ b/bundles/model/org.eclipse.smarthome.model.thing.tests/src/org/eclipse/smarthome/model/thing/tests/GenericItemChannelLinkProviderJavaTest.java
@@ -64,6 +64,8 @@ public class GenericItemChannelLinkProviderJavaTest extends JavaOSGiTest {
 
     @Before
     public void setUp() {
+        registerVolatileStorageService();
+
         initMocks(this);
 
         thingRegistry = getService(ThingRegistry.class);

--- a/bundles/model/org.eclipse.smarthome.model.thing.tests/src/org/eclipse/smarthome/model/thing/tests/GenericItemChannelLinkProviderTest.groovy
+++ b/bundles/model/org.eclipse.smarthome.model.thing.tests/src/org/eclipse/smarthome/model/thing/tests/GenericItemChannelLinkProviderTest.groovy
@@ -21,89 +21,90 @@ import org.eclipse.smarthome.core.thing.ThingRegistry
 import org.eclipse.smarthome.core.thing.link.ItemChannelLinkProvider;
 import org.eclipse.smarthome.core.thing.link.ItemChannelLinkRegistry
 import org.eclipse.smarthome.model.core.ModelRepository
-import org.eclipse.smarthome.model.thing.internal.GenericItemChannelLinkProvider;
 import org.eclipse.smarthome.test.OSGiTest
 import org.junit.After
 import org.junit.Before
 import org.junit.Test;
 
 class GenericItemChannelLinkProviderTest extends OSGiTest {
-	
-	private final static String THINGS_TESTMODEL_NAME = "test.things"
-	private final static String ITEMS_TESTMODEL_NAME = "test.items"
-	
-	ModelRepository modelRepository
-	ThingRegistry thingRegistry
-	ItemRegistry itemRegistry
-	ItemChannelLinkRegistry itemChannelLinkRegistry
+
+    private final static String THINGS_TESTMODEL_NAME = "test.things"
+    private final static String ITEMS_TESTMODEL_NAME = "test.items"
+
+    ModelRepository modelRepository
+    ThingRegistry thingRegistry
+    ItemRegistry itemRegistry
+    ItemChannelLinkRegistry itemChannelLinkRegistry
     ItemChannelLinkProvider itemChannelLinkProvider
-	
-	@Before
-	void setUp() {
-		thingRegistry = getService ThingRegistry
-		assertThat thingRegistry, is(notNullValue())
-		modelRepository = getService ModelRepository
-		assertThat modelRepository, is(notNullValue())
-		itemRegistry = getService ItemRegistry
-		assertThat itemRegistry, is(notNullValue())
-		itemChannelLinkRegistry = getService ItemChannelLinkRegistry
-		assertThat itemChannelLinkRegistry, is(notNullValue())
+
+    @Before
+    void setUp() {
+        registerVolatileStorageService();
+
+        thingRegistry = getService ThingRegistry
+        assertThat thingRegistry, is(notNullValue())
+        modelRepository = getService ModelRepository
+        assertThat modelRepository, is(notNullValue())
+        itemRegistry = getService ItemRegistry
+        assertThat itemRegistry, is(notNullValue())
+        itemChannelLinkRegistry = getService ItemChannelLinkRegistry
+        assertThat itemChannelLinkRegistry, is(notNullValue())
         itemChannelLinkProvider = getService ItemChannelLinkProvider
         assertThat itemChannelLinkProvider, is(notNullValue())
-		modelRepository.removeModel(THINGS_TESTMODEL_NAME)
-		modelRepository.removeModel(ITEMS_TESTMODEL_NAME)
-	}
-	
-	@After
-	void tearDown() {
-		modelRepository.removeModel(THINGS_TESTMODEL_NAME)
-		modelRepository.removeModel(ITEMS_TESTMODEL_NAME)
-	}
+        modelRepository.removeModel(THINGS_TESTMODEL_NAME)
+        modelRepository.removeModel(ITEMS_TESTMODEL_NAME)
+    }
 
-	@Test
-	void 'test that an ItemChannelLink was created for a thing and an item'() {
-		def things = thingRegistry.all
-		assertThat things.size(), is(0)
+    @After
+    void tearDown() {
+        modelRepository.removeModel(THINGS_TESTMODEL_NAME)
+        modelRepository.removeModel(ITEMS_TESTMODEL_NAME)
+    }
 
-		String thingsModel =
-			'''
+    @Test
+    void 'test that an ItemChannelLink was created for a thing and an item'() {
+        def things = thingRegistry.all
+        assertThat things.size(), is(0)
+
+        String thingsModel =
+                '''
 			Bridge hue:bridge:huebridge [ ipAddress = "192.168.3.84", userName = "19fc3fa6fc870a4280a55f21315631f" ] {
 				LCT001 bulb3 [ lightId = "3"	]
 				LCT001 bulb4 [ lightId = "3" ]
 			}
 			'''
-		modelRepository.addOrRefreshModel(THINGS_TESTMODEL_NAME, new ByteArrayInputStream(thingsModel.bytes))
-		def actualThings = thingRegistry.all
+        modelRepository.addOrRefreshModel(THINGS_TESTMODEL_NAME, new ByteArrayInputStream(thingsModel.bytes))
+        def actualThings = thingRegistry.all
 
-		assertThat actualThings.size(), is(3)
-		
-		def items = itemRegistry.items
-		assertThat items.size(), is(0)
-		
-		def itemChannelLinks = itemChannelLinkRegistry.all
-		assertThat itemChannelLinks.size(), is(0)
+        assertThat actualThings.size(), is(3)
 
-		String itemsModel =
-			'''
+        def items = itemRegistry.items
+        assertThat items.size(), is(0)
+
+        def itemChannelLinks = itemChannelLinkRegistry.all
+        assertThat itemChannelLinks.size(), is(0)
+
+        String itemsModel =
+                '''
 			Color	Light3Color					"Light3 Color"		{ channel="hue:LCT001:huebridge:bulb3:color" }
 			'''
-		modelRepository.addOrRefreshModel(ITEMS_TESTMODEL_NAME, new ByteArrayInputStream(itemsModel.bytes))
-		def actualItems = itemRegistry.items
-		
-		assertThat actualItems.size(), is(1)
-			
-		def actualItemChannelLinks = itemChannelLinkRegistry.all
-		assertThat actualItemChannelLinks.size(), is(1)
-		assertThat actualItemChannelLinks.first().toString(), is(equalTo("Light3Color -> hue:LCT001:huebridge:bulb3:color"))
-	}
-	
+        modelRepository.addOrRefreshModel(ITEMS_TESTMODEL_NAME, new ByteArrayInputStream(itemsModel.bytes))
+        def actualItems = itemRegistry.items
+
+        assertThat actualItems.size(), is(1)
+
+        def actualItemChannelLinks = itemChannelLinkRegistry.all
+        assertThat actualItemChannelLinks.size(), is(1)
+        assertThat actualItemChannelLinks.first().toString(), is(equalTo("Light3Color -> hue:LCT001:huebridge:bulb3:color"))
+    }
+
     @Test
     void 'test that multiple links are created'() {
         def things = thingRegistry.all
         assertThat things.size(), is(0)
 
         String thingsModel =
-            '''
+                '''
             Bridge hue:bridge:huebridge [ ipAddress = "192.168.3.84", userName = "19fc3fa6fc870a4280a55f21315631f" ] {
                 LCT001 bulb3 [ lightId = "3" ]
                 LCT001 bulb4 [ lightId = "3" ]
@@ -113,22 +114,22 @@ class GenericItemChannelLinkProviderTest extends OSGiTest {
         def actualThings = thingRegistry.all
 
         assertThat actualThings.size(), is(3)
-        
+
         def items = itemRegistry.items
         assertThat items.size(), is(0)
-        
+
         def itemChannelLinks = itemChannelLinkRegistry.all
         assertThat itemChannelLinks.size(), is(0)
 
         String itemsModel =
-            '''
+                '''
             Color   Light3Color                 "Light3 Color"      { channel="hue:LCT001:huebridge:bulb3:color, hue:LCT001:huebridge:bulb4:color" }
             '''
         modelRepository.addOrRefreshModel(ITEMS_TESTMODEL_NAME, new ByteArrayInputStream(itemsModel.bytes))
         def actualItems = itemRegistry.items
-        
+
         assertThat actualItems.size(), is(1)
-            
+
         def actualItemChannelLinks = itemChannelLinkRegistry.all
         assertThat actualItemChannelLinks.size(), is(2)
         assertThat actualItemChannelLinks.first().toString(),   is(equalTo("Light3Color -> hue:LCT001:huebridge:bulb3:color"))

--- a/bundles/model/org.eclipse.smarthome.model.thing.tests/src/org/eclipse/smarthome/model/thing/tests/GenericThingProviderTest4.groovy
+++ b/bundles/model/org.eclipse.smarthome.model.thing.tests/src/org/eclipse/smarthome/model/thing/tests/GenericThingProviderTest4.groovy
@@ -107,6 +107,8 @@ class GenericThingProviderTest4 extends OSGiTest{
 
     @Before
     public void setUp() {
+        registerVolatileStorageService();
+
         readyService = getService ReadyService
         assertThat readyService, is(notNullValue())
         thingRegistry = getService ThingRegistry

